src/middlewares/delete-project.middleware.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import { remote } from 'electron';
+import {
+  SHOW_DELETE_PROJECT_PROMPT,
+  DELETE_PROJECT_FROM_DISK,
+  FINISH_DELETING_PROJECT_FROM_DISK,
+  deleteProjectFromDisk,
+  deleteProjectFromDiskFinish,
+  refreshProjects,
+  selectProject,
+  createNewProjectStart,
+} from '../actions';
+
+import { getProjectsArray } from '../reducers/projects.reducer';
+
+const { dialog, shell } = remote;
+
+export default (store: any) => (next: any) => (action: any) => {
+  // eslint-disable-next-line default-case
+  switch (action.type) {
+    case SHOW_DELETE_PROJECT_PROMPT: {
+      dialog.showMessageBox(
+        {
+          type: 'warning',
+          buttons: ['Delete from Disk', 'Cancel'],
+          defaultId: 0,
+          title: `Delete ${action.project.name}`,
+          message: `Are you sure you want to delete ${action.project.name}?`,
+          detail: 'WARNING! Deleting from disk will send the project to trash!',
+        },
+        (response: number) => {
+          // TODO: Eventually need to do shouldDeleteFromGuppy as well
+          const shouldDeleteFromDisk = response === 0;
+
+          if (shouldDeleteFromDisk) {
+            const { project } = action;
+            store.dispatch(deleteProjectFromDisk(project));
+          }
+        }
+      );
+      break;
+    }
+
+    case DELETE_PROJECT_FROM_DISK: {
+      const { project } = action;
+      const { path } = project;
+
+      // Returns true if successfully deleted
+      const successfullyDeleted = shell.moveItemToTrash(path);
+
+      if (successfullyDeleted) {
+        store.dispatch(deleteProjectFromDiskFinish(project));
+        store.dispatch(refreshProjects());
```

Comment:
I think dispatching `refreshProjects` is not needed here if you're adding the case `FINISH_DELETING_PROJECT_FROM_DISK` to `projects.reducer` and remove the deleted project from state there.

I think I had it similar in project configuration change and refresh created some troubles - it's possible that this will improve the selection issue.